closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,475 @@
+OPENJ9_FULL_COMPANY_NAME := IBM Corp
+OPENJ9_JAVADOC_BOTTOM = \
+    <a href="$(OPENJ9_BASE_URL)">Link</a> to Eclipse OpenJ9 documentation.<br> \
+    To raise a bug report or suggest an improvement create an <a href="$(OPENJ9_BUG_SUBMIT_URL)">Eclipse Openj9 issue.</a><br> \
+    Copyright &copy; $(FIRST_COPYRIGHT_YEAR), $(COPYRIGHT_YEAR), $(OPENJ9_FULL_COMPANY_NAME). All rights reserved.
+
+#############################################################
+#
+# openj9-sharedclassesdocs
+#
+
+OPENJ9_SHAREDCLASSES_PKGDIR := $(OPENJ9_TOPDIR)/jcl/src/openj9.sharedclasses/share/classes
+OPENJ9_SHAREDCLASSES_PKGS := com.ibm.oti.shared
+
+OPENJ9_SHAREDCLASSES_FIRST_COPYRIGHT_YEAR := 1998
+
+OPENJ9_SHAREDCLASSES_DOCDIR := $(PLATFORM_DOCSDIR)/sharedclasses
+ifneq (,$(findstring openj9-docs-only,$(ALL_OTHER_TARGETS)))
+OPENJ9_SHAREDCLASSES2COREAPI := $(JAVASE_BASE_URL)
+else
+OPENJ9_SHAREDCLASSES2COREAPI := $(JDKJRE2COREAPI)
+endif
+OPENJ9_SHAREDCLASSES_DOCTITLE := OpenJ9 Shared Classes
+OPENJ9_SHAREDCLASSES_WINDOWTITLE := OpenJ9 Shared Classes
+OPENJ9_SHAREDCLASSES_HEADER := <strong>OpenJ9 Shared Classes</strong>
+#OPENJ9_SHAREDCLASSES_OVERVIEW := $(OPENJ9_SHAREDCLASSES_PKGDIR)/sharedclasses-overview.html  # Currently no overview.html for sharedclasses
+
+# The index.html, options, and packages files
+OPENJ9_SHAREDCLASSES_INDEX_FILE := $(OPENJ9_SHAREDCLASSES_DOCDIR)/index.html
+OPENJ9_SHAREDCLASSES_OPTIONS_FILE := $(DOCSTMPDIR)/openj9.sharedclasses.options
+OPENJ9_SHAREDCLASSES_PACKAGES_FILE := $(DOCSTMPDIR)/openj9.sharedclasses.packages
+
+openj9-sharedclassesdocs: $(OPENJ9_SHAREDCLASSES_INDEX_FILE)
+
+# Set relative location to platform document root
+$(OPENJ9_SHAREDCLASSES_INDEX_FILE): GET2DOCSDIR := $(PLATFORM_DOCSDIR)/..
+
+# Run javadoc if the index file is out of date or missing
+$(OPENJ9_SHAREDCLASSES_INDEX_FILE): $(OPENJ9_SHAREDCLASSES_OPTIONS_FILE) $(OPENJ9_SHAREDCLASSES_PACKAGES_FILE) coredocs
+	$(prep-javadoc)
+	$(call JavadocSummary,$(OPENJ9_SHAREDCLASSES_OPTIONS_FILE),$(OPENJ9_SHAREDCLASSES_PACKAGES_FILE))
+	$(JAVADOC_CMD) -d $(@D) \
+	    @$(OPENJ9_SHAREDCLASSES_OPTIONS_FILE) @$(OPENJ9_SHAREDCLASSES_PACKAGES_FILE)
+
+$(OPENJ9_SHAREDCLASSES_OPTIONS_FILE): FIRST_COPYRIGHT_YEAR=$(OPENJ9_SHAREDCLASSES_FIRST_COPYRIGHT_YEAR)
+$(OPENJ9_SHAREDCLASSES_OPTIONS_FILE):
+	$(prep-target)
+	echo COREAPI_DOCSDIR is $(COREAPI_DOCSDIR)
+	echo OPENJ9_SHAREDCLASSES2COREAPI is $(OPENJ9_SHAREDCLASSES2COREAPI)
+	@($(call OptionOnly,$(COMMON_JAVADOCFLAGS)) ; \
+	  $(call OptionPair,-sourcepath,$(OPENJ9_DOCS_SRCDIR)) ; \
+	  $(call OptionPair,-encoding,ascii) ; \
+	  $(call OptionPair,-doctitle,$(OPENJ9_SHAREDCLASSES_DOCTITLE)) ; \
+	  $(call OptionPair,-windowtitle,$(OPENJ9_SHAREDCLASSES_WINDOWTITLE) $(DRAFT_WINTITLE)); \
+	  $(call OptionPair,-header,$(OPENJ9_SHAREDCLASSES_HEADER)$(DRAFT_HEADER)) ; \
+	  $(call OptionPair,-bottom,$(OPENJ9_JAVADOC_BOTTOM)$(DRAFT_BOTTOM)) ; \
+	  $(call OptionTrip,-linkoffline,$(JAVASE_BASE_URL),$(COREAPI_DOCSDIR)/); \
+	) >> $@
```

Comment:
Why does this append to the target file rather than just write to it (`>>` instead of `>`)?

Reply:
Just because that's what the similar targets in make/Javadoc.gmk do.  Doesn't alter the output to use '>', so I've changed it for the targets in this file.